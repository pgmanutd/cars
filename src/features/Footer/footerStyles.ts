import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      borderTop: `1px solid ${theme.palette.divider}`,
      paddingTop: theme.spacing(3),
      paddingBottom: theme.spacing(3),
    },
  }),
);
