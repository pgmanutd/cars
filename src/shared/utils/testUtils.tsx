import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { render } from '@testing-library/react';

export const renderWithRouter = (
  ui: React.ReactElement,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  }: { route?: string; history?: MemoryHistory } = {},
) => ({
  ...render(<Router history={history}>{ui}</Router>),
  history,
});
