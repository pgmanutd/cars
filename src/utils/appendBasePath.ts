const appendBasePath = (basePath: string, path: string) =>
  `${basePath}?${path}`;

export default appendBasePath;
