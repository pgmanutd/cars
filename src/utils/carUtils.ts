import { Car } from 'features/CarInfo/types';

export const getCarName = ({ manufacturerName, modelName }: Car) =>
  `${manufacturerName} ${modelName}`;

export const getCarFeatures = (
  { stockNumber, mileage, fuelType, color }: Car,
  { stockNumberPrefix = '' }: { stockNumberPrefix?: string } = {
    stockNumberPrefix: '',
  },
) =>
  `${stockNumberPrefix}${stockNumber} - ${mileage?.number} ${mileage?.unit} - ${fuelType} - ${color}`;
