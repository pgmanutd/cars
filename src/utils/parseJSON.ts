const parseJSON = <T>(stringifiedJSON: string | null, fallbackValue: T) => {
  try {
    if (!stringifiedJSON) {
      return fallbackValue;
    }

    return JSON.parse(stringifiedJSON);
  } catch (e) {
    return fallbackValue;
  }
};

export default parseJSON;
