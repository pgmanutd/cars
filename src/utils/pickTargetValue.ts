import React from 'react';

const pickTargetValue = (event: React.ChangeEvent<{ value: unknown }>) =>
  event.target.value;

export default pickTargetValue;
