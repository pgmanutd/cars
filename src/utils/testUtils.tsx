import React from 'react';
import { Router, Route } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { HelmetProvider } from 'react-helmet-async';
import { render } from '@testing-library/react';

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    routerConfig: {
      path = '/',
      route = '/',
      history = createMemoryHistory({ initialEntries: [route] }),
    },
  }: {
    routerConfig: { path?: string; route?: string; history?: MemoryHistory };
  } = { routerConfig: {} },
) => ({
  ...render(
    <HelmetProvider>
      <Router history={history}>
        <Route
          path={path}
          render={routeProps => React.cloneElement(ui, { ...routeProps })}
        />
      </Router>
    </HelmetProvider>,
  ),
  history,
});
