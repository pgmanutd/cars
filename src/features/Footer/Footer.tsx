import React from 'react';
import Typography from '@material-ui/core/Typography';
import Link from '@material-ui/core/Link';
import Box from '@material-ui/core/Box';

import routePaths from 'constants/routePaths';

import { useTranslate } from 'features/Translate';

import { useStyles } from './footerStyles';

const Footer: React.FC = () => {
  const classes = useStyles();
  const { translate } = useTranslate();

  return (
    <Box data-testid="Footer" component="footer" className={classes.root}>
      <Typography variant="body2" color="textPrimary" align="center">
        {translate('features.Footer.copyright')}{' '}
        <Link color="inherit" href={routePaths.cars}>
          {translate('features.Footer.homeLinkText')}
        </Link>{' '}
        {new Date().getFullYear()}.
      </Typography>
    </Box>
  );
};

export default Footer;
