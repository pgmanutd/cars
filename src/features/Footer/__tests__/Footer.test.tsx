import React from 'react';

import { renderWithProviders } from 'utils/testUtils';

import Footer from '../Footer';

describe('<Footer />', () => {
  const setup = () => {
    const { renderResult } = renderWithProviders(<Footer />);

    return {
      renderResult,
    };
  };

  it('should render the component and matches it against stored snapshot', () => {
    const { renderResult } = setup();

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
});
