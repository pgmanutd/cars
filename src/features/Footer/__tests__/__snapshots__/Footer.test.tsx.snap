// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <footer
    class="MuiBox-root MuiBox-root-3 makeStyles-root-1"
    data-testid="Footer"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
    >
      Â© 
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="/"
      >
        Cars
      </a>
       2020.
    </p>
  </footer>
</DocumentFragment>
`;
